src/citationnet/static/force-citationNet.js

```diff
@@ -0,0 +1,419 @@
+
+            // return minStrength if out smaller than minStrength
+            // return maxStrength if out larger than maxStrength
+            // return out **
+            return out <= minStrength ? minStrength
+                : out >= maxStrength ? maxStrength
+                    : out ** exp;
+        };
+        return strengthFunc;
+    }
+
+    /**
+    * Preprocess this.data
+    */
+    processData() {
+        var data = this.data;
+        var id_map = {};
+
+        // find input node
+        this.inputNode = data.nodes.filter(o => o.attributes.is_input_DOI == true)[0];
+        var inputNode = this.inputNode;
+
+        for (let i = 0; i < data.nodes.length; i++) {
+            id_map[data.nodes[i].id] = i;
+            data.nodes[i].outgoingLinks = [];
+            data.nodes[i].outgoingLinkTo = [];
+            data.nodes[i].incomingLinks = [];
+            data.nodes[i].incomingLinkFrom = [];
+
+            // delete unused attributes
+            delete data.nodes[i].color;
+            delete data.nodes[i].size;
+            delete data.nodes[i].x;
+            delete data.nodes[i].y;
+
+            // fix z-coordinate of nodes depending on publication year
+            // 20 units between input node and years before/after
+            // 5 units spacing between years
+            if (data.nodes[i].attributes.nodeyear >= inputNode.attributes.nodeyear) {
+                data.nodes[i].fz = 5 * (data.nodes[i].attributes.nodeyear - inputNode.attributes.nodeyear + 20);
+            } else {
+                data.nodes[i].fz = 5 * (data.nodes[i].attributes.nodeyear - inputNode.attributes.nodeyear - 20);
+            }
+        }
+
+        // fix position of input node, color red
+        inputNode.fx = 0.0;
+        inputNode.fy = 0.0;
+        inputNode.fz = 0.0;
+        inputNode.color = 'red';
+
+        // cross-link node objects
+        data.edges.forEach(edge => {
+            var a = data.nodes[id_map[edge.source]];
+            var b = data.nodes[id_map[edge.target]];
+            if (typeof a != "undefined" && typeof b != "undefined") {
+              !a.outgoingLinks && (a.outgoingLinks = []);
+              a.outgoingLinks.push(edge);
+
+              !a.outgoingLinkTo && (a.outgoingLinkTo = []);
+              a.outgoingLinkTo.push(b);
+
+              !b.incomingLinks && (b.incomingLinks = []);
+              b.incomingLinks.push(edge);
+
+              !b.incomingLinkFrom && (b.incomingLinkFrom = []);
+              b.incomingLinkFrom.push(a);
+
+              delete edge.color;
+              delete edge.size;
+            }
+        }
+      );
+    }
+
+    /**
+    * Move camera to default view point. Triggered by UI.
+    * @param {String} viewPoint - either "top" or "side"
+    * @returns {ReturnValueDataTypeHere} Brief description of the returning value here.
+    */
+    view(viewPoint) {
+        if (viewPoint == 'top') {
```

Comment:
Not a big deal, but you may want to be more consistent between single and double quotes for string literals (pick one)